Assets/UI/launchbar.lua

```diff
@@ -574,57 +539,77 @@ function RefreshView()
 end
 
 -- ===========================================================================
-function OnTurnBegin()
-  local localPlayer = Players[Game.GetLocalPlayer()];
-  if (localPlayer == nil) then
-    return;
-  end
-  local playerTechs = localPlayer:GetTechs();
-  local currentTechID:number = playerTechs:GetResearchingTech();
-
-  if(currentTechID >= 0) then
-    local progress:number = playerTechs:GetResearchProgress(currentTechID);
-    local cost:number = playerTechs:GetResearchCost(currentTechID);
+function UpdateTechMeter( localPlayer:table )
+  if ( localPlayer ~= nil and Controls.ScienceHookWithMeter:IsVisible() ) then
+    local playerTechs          = localPlayer:GetTechs();
+    local currentTechID:number = playerTechs:GetResearchingTech();
+    if(currentTechID >= 0) then
+      local progress:number = playerTechs:GetResearchProgress(currentTechID);
+      local cost:number = playerTechs:GetResearchCost(currentTechID);
+
+      Controls.ScienceMeter:SetPercent(progress/cost);
+      Controls.ScienceTurnsLeft:SetText(playerTechs:GetTurnsLeft());
```

Comment:
ScienceTurnsLeft does not exist anymore, needs to be removed here and just below